falafel/core/evaluators.py

```diff
@@ -335,15 +335,22 @@ def get_contextual_hostname(self, default=""):
         return self.system_id
 
     def get_branch_info(self):
+        version = None;
+        hostname = None;
         branch_info = json.loads(self.spec_mapper.get_content("branch_info",
                                  split=False, default="{}"))
+        product = branch_info.get("product")
         remote_branch = branch_info.get("remote_branch")
         if remote_branch == -1:
             remote_branch = None
         remote_leaf = branch_info.get("remote_leaf")
         if remote_leaf == -1:
             remote_leaf = None
-        return remote_branch, remote_leaf
+        if hasattr(product, "type") and product["type"] == "Satellite":
```

Comment:
Sorry, I assumed it was a `dict` from the usage so I should have dug in further.